### PR TITLE
Fix InvertedLog10Transform.inverted() 

### PR DESCRIPTION
**Pull request by fkloosterman**
## PR Summary

Closes issue #10202 

InvertedLog10Transform.inverted() method fails because position argument `nonpos` is missing. To fix this, a default value for the `nonpos` argument is added in constructor of `LogTransformBase` base class (and to be consistent, also to `LogitTransform` and `LogisticTransform` constructors). While testing it was noted that repr of Log/Logit/LogisticTransform objects results in max recursion error, because the `Transform` base class implements `__repr__` with a call to `str(self)` and no `__str__` methods are implemented for the Log/Logit/LogisticTransform classes. This was fixed as well.

## PR Checklist

- [x] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by anntzer** [comment]
Fixng a RecursionError seems release-critical...

File: lib/matplotlib/scale.py [hunk 1]
```diff
@@ -93,7 +93,7 @@ class LogTransformBase(Transform):
     is_separable = True
     has_inverse = True
 
-    def __init__(self, nonpos):
+    def __init__(self, nonpos='mask'):
```
Comment (tacaswell) [inline lib/matplotlib/scale.py]: This should probably default to 'clip' as this is the default else where.

Comment (fkloosterman) [reply lib/matplotlib/scale.py]: You are right, LogScale uses clip as default and so should LogTransform. Fixed in the PR.

**Comment by fkloosterman** [comment]
@dstansby: I made the changes to conform to PEP8 and all checks pass now.

File: lib/matplotlib/scale.py [hunk 2]
```diff
@@ -448,7 +455,7 @@ class LogitTransform(Transform):
...
```
Comment (dstansby) [inline lib/matplotlib/scale.py]: I think this needs changing to 'clip' instead of 'mask'

Comment (fkloosterman) [reply lib/matplotlib/scale.py]: For consistency, I used the same default for LogitTransform (and LogisticTransform) as for LogitScale, which already uses 'mask'.

Comment (dstansby) [reply lib/matplotlib/scale.py]: 👍

File: lib/matplotlib/tests/test_scale.py [hunk 3]
```diff
@@ -74,6 +74,19 @@ def test_extra_kwargs_raise():
...
+    fig, ax = plt.subplots()
+    ax.set_yscale('log')
+    # get transformation from data to axes
+    tform = (ax.transAxes + ax.transData.inverted()).inverted()
+
+
+def test_logscale_transform_repr():
+    fig, ax = plt.subplots()
+    ax.set_yscale('log')
+    s = repr(ax.transData)
```
Comment (phobson) [inline lib/matplotlib/tests/test_scale.py]: show we `assert` anything there?

Comment (jklymak) [reply lib/matplotlib/tests/test_scale.py]: ping @fkloosterman - can this test do more than just make sure it runs?

Comment (fkloosterman) [reply lib/matplotlib/tests/test_scale.py]: you mean something like:
`assert repr(matplotlib.scale.Log10Transform(nonpos='clip'))=="Log10Transform('clip')"`?

Comment (phobson) [reply lib/matplotlib/tests/test_scale.py]: @fkloosterman yes something like that. Note, the pep8 checker will complain about the last line unless you put spaces around the `==` operator:
`assert s == "Log10Transform('clip')"`

**Comment by jklymak** [comment]
@phobson  is this one OK by you now?

**Comment by phobson** [comment]
@jklymak Looks good to me. Travis failure seems unrelated, yeah?

**Comment by jklymak** [comment]
Yeah, I think sometimes Travis has a minor problem with tex...

**Comment by jklymak** [comment]
Thanks a lot @fkloosterman!
